File: Sorting-Algorithm/bubblesort/bubbleSort.js
// bubble-Sort in JavaScript

function bubbleSort(ar) {
    let temp;
    for (let i = 0; i < ar.length-1; i++) {
        for (let j = 0; j < ar.length-i-1; j++) {
            if (ar[j]>ar[j+1]) {
                // [ar[j],ar[j+1]]=[ar[j+1],ar[j]]  -->can use this swapping technique as well
                temp=ar[j] ;
                ar[j]=ar[j+1];
                ar[j+1]=temp

            }            
        }
        
    }
    return ar;
}
arr=[5,4,3,2,1];
console.log("array before sorting:",arr)
swapped_array=bubbleSort(arr);
console.log("array before sorting:",swapped_array);

File: Sorting-Algorithm/bubblesort/bubble_sort.py
# day 1
# bubble sort
#  time complexity = O(n2)

def bubble_sort(a):
    for i in range(len(arr)-1):
        for j in range(len(arr)-1-i):
            if a[j]>a[j+1]:
                a[j],a[j+1]=a[j+1],a[j]
    return a

# driver code
arr=[9,8,76,5,4]
print("Before sorting",arr)
print("After Sorting",bubble_sort(arr))



